gold-phone-input.html

```diff
@@ -181,13 +181,19 @@
 
         formattedValue += value[i];
       }
-
       this.value = formattedValue.trim();
 
+      // The cursor automatically jumps to the end after re-setting the value,
+      // so restore it to its original position.
+      this.$.input.selectionStart = start;
+      this.$.input.selectionEnd = start;
+
       // If the character right before the selection is a newly inserted
       // dash, we need to advance the selection to maintain the caret position.
-      if (!previousCharADash && this.value.charAt(start - 1) == '-')
-        this.$.input.selectionStart++;
+      if (!previousCharADash && this.value.charAt(start - 1) == '-') {
+        this.$.input.selectionStart = start + 1;
+        this.$.input.selectionEnd = start + 1;
+      }
```

Comment:
Not sure. Maybe a `gold-formatting-behaviour` that has a divider/how often properties? The problem is that the `this.value = ` assignment would be outside whatever helper function the behaviour was, and that's the one that throws the caret off :(